localstack/services/secretsmanager/provider.py

```diff
@@ -0,0 +1,266 @@
+from localstack.aws.api import RequestContext
+from localstack.aws.api.secretsmanager import (
+    AddReplicaRegionListType,
+    BooleanType,
+    CancelRotateSecretResponse,
+    ClientRequestTokenType,
+    CreateSecretResponse,
+    DeleteResourcePolicyResponse,
+    DeleteSecretResponse,
+    DescribeSecretResponse,
+    DescriptionType,
+    ExcludeCharactersType,
+    ExcludeLowercaseType,
+    ExcludeNumbersType,
+    ExcludePunctuationType,
+    ExcludeUppercaseType,
+    FiltersListType,
+    GetRandomPasswordResponse,
+    GetResourcePolicyResponse,
+    GetSecretValueResponse,
+    IncludeSpaceType,
+    KmsKeyIdType,
+    ListSecretsResponse,
+    ListSecretVersionIdsResponse,
+    MaxResultsType,
+    NameType,
+    NextTokenType,
+    NonEmptyResourcePolicyType,
+    PasswordLengthType,
+    PutResourcePolicyResponse,
+    PutSecretValueResponse,
+    RecoveryWindowInDaysType,
+    RemoveRegionsFromReplicationResponse,
+    RemoveReplicaRegionListType,
+    ReplicateSecretToRegionsResponse,
+    RequireEachIncludedTypeType,
+    RestoreSecretResponse,
+    RotateSecretResponse,
+    RotationLambdaARNType,
+    RotationRulesType,
+    SecretBinaryType,
+    SecretIdType,
+    SecretsmanagerApi,
+    SecretStringType,
+    SecretVersionIdType,
+    SecretVersionStagesType,
+    SecretVersionStageType,
+    SortOrderType,
+    StopReplicationToReplicaResponse,
+    TagKeyListType,
+    TagListType,
+    UpdateSecretResponse,
+    UpdateSecretVersionStageResponse,
+    ValidateResourcePolicyResponse,
+)
+from localstack.services.moto import call_moto
+from localstack.services.secretsmanager.secretsmanager_patches import apply_patches
+
+
+class SecretsmanagerProvider(SecretsmanagerApi):
+    def __init__(self):
+        apply_patches()
+
+    def cancel_rotate_secret(
+        self, context: RequestContext, secret_id: SecretIdType
+    ) -> CancelRotateSecretResponse:
+        raise NotImplementedError
+
+    def create_secret(
+        self,
+        context: RequestContext,
+        name: NameType,
+        client_request_token: ClientRequestTokenType = None,
+        description: DescriptionType = None,
+        kms_key_id: KmsKeyIdType = None,
+        secret_binary: SecretBinaryType = None,
+        secret_string: SecretStringType = None,
+        tags: TagListType = None,
+        add_replica_regions: AddReplicaRegionListType = None,
+        force_overwrite_replica_secret: BooleanType = None,
+    ) -> CreateSecretResponse:
+        res = call_moto(context)
```

Comment:
In fact, I believe we could remove the methods that are only forwarding the request verbatim to moto (without applying any patches to the requests/responses), as `MotoFallbackDispatcher` should already take care of this transparently. 👍